### PR TITLE
Require username

### PR DESCRIPTION
**Pull request by paymog**
Force all fields to be filled out before logging in or creating an account.

### CODE REVIEW

**Comment by kjrcda** [comment]
Made the file name changes but had to do it on GitHub site. Not sure how to do it with the folder name though...

**Comment by paymog** [comment]
Josh, when you're commenting, do you use the "Files Changed" tab or the "Commits" tab at the top of this page?
screen shot 2015-02-13 at 10 48 40 am

File: webapp/ng_tests/controllers/main_page_controller_test.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            $httpBackend.expectPOST('/api/users/authenticate').respond(500, errorMessage);
+            $httpBackend.flush();
+
+            expect($rootScope.errorMessages).toBe(errorMessage);
+            expect($location.path()).toBe('/')
+        });
+
+        it('does not make post without email', function () {
+            createController();
+            $rootScope.email = DEFAULT_PASSWORD;
```
Comment (BrandonWade) [inline webapp/ng_tests/controllers/main_page_controller_test.js]: Isn't this test supposed to check a post without an email? This sets the email to the password

File: webapp/ng_tests/controllers/main_page_controller_test.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+            //note do not expectPOST
+
+            expect($location.path()).toBe('/');
+            expect($rootScope.errorMessages).toBeFalsy();
+        });
+
+        it('does not make post without password', function () {
+            createController();
+            $rootScope.password = DEFAULT_PASSWORD;
```
Comment (BrandonWade) [inline webapp/ng_tests/controllers/main_page_controller_test.js:76]: Same as previous comment; this sets the password. Isn't this test checking for no password?

**Comment by BrandonWade** [comment]
Looks good. Ready to merge.

**Comment by paymog** [comment]
Fuck...this was merged into master

**Pull request merged**